dotnetv3/AutoScaling/scenarios/AutoScale_Basics/AutoScale_Basics/AutoScaleBasics.cs

```diff
@@ -0,0 +1,247 @@
+  2. Creates an AWS Auto Scaling group.
+  3. Shows the details of the new AWS Auto Scaling group to show that only
+     one instance was created.
+  4. Enables metrics collection.
+  5. Updates the AWS Auto Scaling group to increase the max size to three.
+  6. Describes AWS Auto Scaling groups again to show the current state of the
+     group.
+  7. Changes the desired capacity of the AWS Auto Scaling group to two.
+  9. Retrieves the details of the group and shows the number of instances.
+ 10. Lists the scaling activities that have occurred for the group.
+ 11. Displays the Amazon CloudWatch metrics that have been collected.
+ 12. Terminates an instance in the Auto Scaling group.
+ 13. Disables metrics collection.
+ 13. Deletes the Auto Scaling group.
+ 14. Deletes the EC2 launch template.
+*/
+
+// snippet-start:[AutoScale.dotnetv3.AutoScale_Basics.main]
+var imageId = "ami-05803413c51f242b7";
+var instanceType = "t2.micro";
+var launchTemplateName = "AutoScaleLaunchTemplate";
+
+// The name of the Auto Scaling group.
+var groupName = "AutoScaleExampleGroup";
+
+// the Amazon Resource Name (ARN) of the service linked IAM role.
+var serviceLinkedRoleARN = "<Enter Value>";
+
+var client = new AmazonAutoScalingClient(RegionEndpoint.USEast2);
+
+Console.WriteLine("Auto Scaling Basics");
+DisplayDescription();
+
+// Start by creating the and save launch template Id to use when deleting the launch template at
+// the end of the application.
+var launchTemplateId = await EC2Methods.CreateLaunchTemplateAsync(imageId, instanceType, launchTemplateName);
+
+// Confirm that the template was created by asking for a description of it.
+await EC2Methods.DescribeLaunchTemplateAsync(launchTemplateName);
+
+PressEnter();
+
+Console.WriteLine($"--- Creating an Auto Scaling group named {groupName}. ---");
+var success = await AutoScaleMethods.CreateAutoScalingGroup(
+    client,
+    groupName,
+    launchTemplateName,
+    serviceLinkedRoleARN);
+
+// Keep checking the details of the new group until its lifecycle state
+// is "InService"
+Console.WriteLine($"Waiting for the Auto Scaling group to be active.");
+
+List<AutoScalingInstanceDetails> instanceDetails;
+
+do
+{
+    instanceDetails = await AutoScaleMethods.DescribeAutoScalingInstancesAsync(client, groupName);
+}
+while (instanceDetails.Count <= 0);
+
+Console.WriteLine($"Auto scaling group {groupName} successfully created.");
+Console.WriteLine($"{instanceDetails.Count} instances were created for the group.");
+
+// Display the details of the AWS Auto Scaling group.
+instanceDetails.ForEach(detail =>
+{
+    Console.WriteLine($"Group name: {detail.AutoScalingGroupName}");
+});
+
+PressEnter();
+
+Console.WriteLine($"\n--- Enable metrics collection for {groupName}");
+await AutoScaleMethods.EnableMetricsCollectionAsync(client, groupName);
+
+// Show the metrics that are collected for the group.
+
+// Now update the group to allow up to 3 instances to be created for the
+// auto scaling group.
```

Comment:
// Update the maximum size of the group to three instances.

Reply:
changed